providers/providers.go

```diff
@@ -30,7 +30,7 @@ var defaultCleanupInterval = time.Hour
 type ProviderManager struct {
 	// all non channel fields are meant to be accessed only within
 	// the run method
-	providers *lru.Cache
+	providers *lru.LRU
```

Comment:
what's the difference between these two?

Reply:
I guess thread-safety.

Reply:
Are we sure this is safe? It could come back to bite us.

Reply:
Yes. The old provider set wasn't safe to use from multiple threads _anyways_ as we'd modify the `*providerSet` structs.